contrib/drivers/filesharedrivers/oceanstor/oceanclient.go

```diff
@@ -0,0 +1,503 @@
+		"password": c.password,
+		"scope":    "0",
+	}
+
+	url := "/xxxxx/sessions"
+	resp, err := c.request(url, "POST", data)
+	if err != nil {
+		return nil, err
+	}
+
+	var auth Auth
+
+	if err := handleReponse(resp, &auth); err != nil {
+		return nil, err
+	}
+
+	return &auth.Data, nil
+}
+
+type Protocol interface {
+	createShare(fsName, fsID string) (interface{}, error)
+	getShare(fsName string) (interface{}, error)
+	getShareID(share interface{}) string
+	deleteShare(shareID string) error
+	getShareByID(shareID string) (interface{}, error)
+	getLocation(sharePath, ipAddr string) string
+	allowAccess(shareID, accessTo, accessLevel string) (interface{}, error)
+	getAccessLevel(accessLevel string) string
+}
+
+func NewProtocol(proto string, c *Client) Protocol {
+	switch proto {
+	case NFSProto:
+		return &NFS{Client: c}
+	case CIFSProto:
+		return &CIFS{Client: c}
+	}
+
+	return nil
+}
+
+func (c *Client) createFileSystem(name, poolID string, size int64) (*FileSystemData, error) {
+	data := map[string]interface{}{
+		"PARENTID":   poolID,
+		"NAME":       name,
+		"PARENTTYPE": 216,
+		"ALLOCTYPE":  1,
+		"CAPACITY":   Gb2Sector(size),
+	}
+
+	url := "/filesystem"
+	resp, err := c.request(url, "POST", data)
+	if err != nil {
+		return nil, err
+	}
+
+	var fileSystem FileSystem
+
+	if err := handleReponse(resp, &fileSystem); err != nil {
+		return nil, err
+	}
+
+	return &fileSystem.Data, nil
+}
+
+func (c *Client) deleteFS(fsID string) error {
+	url := "/filesystem/" + fsID
+	resp, err := c.request(url, "DELETE", nil)
+	if err != nil {
+		return err
+	}
+
+	var errDelete DeleteError
+	if err := handleReponse(resp, &errDelete); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Client) getAllLogicalPort() ([]LogicalPortData, error) {
+	url := "/LIF"
+	resp, err := c.request(url, "GET", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var logicalPortList LogicalPortList
+
+	if err := handleReponse(resp, &logicalPortList); err != nil {
+		return nil, err
+	}
+
+	return logicalPortList.Data, nil
+}
+
+func (c *Client) getFileSystem(fsid string) (*FileSystemData, error) {
+	url := "/filesystem/" + fsid
+	resp, err := c.request(url, "GET", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var fileSystem FileSystem
+	if err := handleReponse(resp, &fileSystem); err != nil {
+		return nil, err
+	}
+
+	return &fileSystem.Data, nil
+}
+
+func (c *Client) getFileSystemByName(name string) ([]FileSystemData, error) {
+	url := "/filesystem?filter=NAME::" + name
+	resp, err := c.request(url, "GET", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var fsList FileSystemList
+
+	if err := handleReponse(resp, &fsList); err != nil {
+		return nil, err
+	}
+
+	return fsList.Data, nil
+}
+
+func (c *Client) deleteFileSystem(name, poolID string) error {
+	data := map[string]interface{}{
+		"PARENTID":   poolID,
+		"NAME":       name,
+		"PARENTTYPE": 216,
+		"ALLOCTYPE":  1,
+	}
+
+	url := "/filesystem"
+	resp, err := c.request(url, "POST", data)
```

Comment:
Why it's `POST` method here?

Reply:
This method has been deprecated and will be deleted.